### PR TITLE
Version 1.0.0-pre.1

### PR DESCRIPTION
**Pull request by dturn**
**What are you trying to accomplish with this PR?**
Release 1.0.0.pre.1

**How is this accomplished?**
Update version, changelog, and migration guide.

**What could go wrong?**
Something missing from the changelog.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,20 @@
...
+
+- This is the final release of KubernetesDeploy. Version 1.0.0 will be released
+under the name `Krane`. We've added a migration guide to help make it easier to migrate.
+([#607](https://github.com/Shopify/kubernetes-deploy/pull/607))
+
+*Enhancements*
+- (beta) `krane deploy` will now consider all namespaced resources eligible for pruning,
+it does not respect the `krane.shopify.io/prunable` annotation ([#616](https://github.com/Shopify/kubernetes-deploy/pull/616)
+
+- (beta) Added the `--std-in` flag to `krane deploy|global-deploy|render` to read resources from stdin. ([#630](https://github.com/Shopify/kubernetes-deploy/pull/630))
```
Comment (RyanBrushett) [inline CHANGELOG.md]:
```suggestion
- (beta) Added the `--stdin` flag to `krane deploy|global-deploy|render` to read resources from stdin. ([#630](https://github.com/Shopify/kubernetes-deploy/pull/630))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,20 @@
...
+## 1.0.0.pre.1
+
+*Important!*
+
+- This is the final release of KubernetesDeploy. Version 1.0.0 will be released
+under the name `Krane`. We've added a migration guide to help make it easier to migrate.
+([#607](https://github.com/Shopify/kubernetes-deploy/pull/607))
+
+*Enhancements*
+- (beta) `krane deploy` will now consider all namespaced resources eligible for pruning,
```
Comment (RyanBrushett) [inline CHANGELOG.md]: How many does this add? Should we make a note which ones this adds compared to our whitelist as of the time of writing?

File: lib/krane/cluster_resource_discovery.rb [hunk 3]
```diff
@@ -17,7 +17,7 @@ def crds
     end
 
     def prunable_resources(namespaced:)
-      black_list = %w(Namespace Node)
+      black_list = %w(Namespace Node ControllerRevision)
```
Comment (dturn) [inline lib/krane/cluster_resource_discovery.rb]: `ControllerRevision` are created by DS controllers and are use for rollbacks. They get the last-applied annotation and I suspect we dont' want to be pruning them https://kubernetes.io/docs/tasks/manage-daemon/rollback-daemon-set/#understanding-daemonset-revisions.

Comment (RyanBrushett) [reply lib/krane/cluster_resource_discovery.rb]: Good catch

File: 1.0-Upgrade.md [hunk 4]
```diff
@@ -8,6 +8,8 @@
 * The command-line interface was redesigned; alongside the name change there are breaking changes in several flags.
 * There are breaking changes in the public API (such as the renaming of the `KubernetesDeploy` namespace to `Krane`, and the change in default values for different arguments of the public interface).
 * StatsD metrics will now be generated with the `krane` prefix.
+* `krane deploy` now considers all namespaced resources eligible for pruning, including
+custom resources.
```
Comment (RyanBrushett) [inline 1.0-Upgrade.md]:
```suggestion
custom resources. See blacklist for exceptions.
```

and link to blacklist maybe?

File: 1.0-Upgrade.md [hunk 5]
```diff
@@ -8,6 +8,8 @@
 * The command-line interface was redesigned; alongside the name change there are breaking changes in several flags.
 * There are breaking changes in the public API (such as the renaming of the `KubernetesDeploy` namespace to `Krane`, and the change in default values for different arguments of the public interface).
 * StatsD metrics will now be generated with the `krane` prefix.
+* `krane deploy` now considers all namespaced resources eligible for pruning, including
+custom resources. See blacklist for exceptions.
```
Comment (RyanBrushett) [inline 1.0-Upgrade.md]: link pls

Comment (dturn) [reply 1.0-Upgrade.md]: fixed

**Pull request merged**